### ISSUE TITLE
Synctex fails to work.

### ISSUE THREAD

**Issue by joliang17** [James-Yu/LaTeX-Workshop#1749]
**Disable all the other extensions except for LaTeX Workshop, and check that you still see this issue. [Required]**

You still see this issue?: Yes

**Describe the bug [Required]**
When I used Latex in vscode and open pdf with vscode viewer these weeks, I found that I cannot use either "SyncTeX from cursor" or reverse synctex at any time. 

**To Reproduce**
1. Open a .tex file
2. Use "View Latex PDF file" to open the pdf viewer in a new tab.
3. Use "SyncTeX from cursor" or "ctrl+click" to activate the reverse synctex.

**Logs [Required]**

[21:31:25] SYNCTEX command invoked.
[21:31:25] SyncTeX failed.
[21:31:25] SyncTexJsError: synctex file, .synctex and .synctex.gz, not found in the file system.
	at parseSyncTexForPdf (c:\Users\Joanna\.vscode\extensions\james-yu.latex-workshop-8.2.0\out\src\components\synctex.js:23:11)
	at Object.syncTexJsForward (c:\Users\Joanna\.vscode\extensions\james-yu.latex-workshop-8.2.0\out\src\components\synctex.js:48:27)
	at Locator.syncTeX (c:\Users\Joanna\.vscode\extensions\james-yu.latex-workshop-8.2.0\out\src\components\locator.js:126:42)
	at Commander.synctex (c:\Users\Joanna\.vscode\extensions\james-yu.latex-workshop-8.2.0\out\src\commander.js:210:32)
	at vscode.commands.registerCommand (c:\Users\Joanna\.vscode\extensions\james-yu.latex-workshop-8.2.0\out\src\main.js:201:89)
	at v._executeContributedCommand (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:518:753)
	at v.$executeContributedCommand (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:519:104)
	at p._doInvokeHandler (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:635:275)
	at p._invokeHandler (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:634:971)
	at p._receiveRequest (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:633:639)
	at p._receiveOneMessage (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:632:468)
	at _protocol.onMessage.e (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:630:691)
	at l.fire (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:50:519)
	at _.fire (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:181:274)
	at e.onMessage.e (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:787:104)
	at l.fire (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:50:519)
	at _.fire (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:181:274)
	at t.PersistentProtocol._receiveMessage (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:185:629)
	at _socketDisposables.push._socketReader.onMessage.e (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:182:824)
	at l.fire (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:50:519)
	at p.acceptChunk (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:178:588)
	at _register._socket.onData.e (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:177:958)
	at Socket.t (e:\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:187:68)
	at Socket.emit (events.js:182:13)
	at addChunk (_stream_readable.js:283:12)
	at readableAddChunk (_stream_readable.js:264:11)
	at Socket.Readable.push (_stream_readable.js:219:10)
	at Pipe.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)

**Desktop [Required]:**
 - OS: Windows 10
 - VS Code version: 1.39.1
 - Extension version: 8.2.0

**Are you using LaTeX Workshop with VS Code Remote?**

No

**Comment by jlelong** [comment]
I confirm that synctex works so this has to be a configuration issue on your side.
Do you have a `.synctex` file next to your `.tex` file?

**Comment by joliang17** [comment]
No. 
![image](https://user-images.githubusercontent.com/47235014/66799856-a8d5d780-eec7-11e9-8d67-ab14072ecf20.png)
Meanwhile, in my latex workshop settings, .synctex file is not included in the "Clean: File Type", so I don't know why synctex file cannot be found in file system.

**Comment by jlelong** [comment]
Weird. Can you show us the full logs of the LaTeX compiler?

**Comment by joliang17** [comment]
[21:52:57] Parsing d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[21:52:57] File watcher: responding to change in d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[21:52:57] d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex changed. Auto build project.
[21:52:57] BUILD command invoked.
[21:52:57] Building root file: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[21:52:57] Build root file d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[21:52:57] Recipe step 1: pdflatex, --max-print-line=10000,-synctex=1,-interaction=nonstopmode,-file-line-error,d:/Google Drive/UCB/Fall 2019/242/hw3/Homework 3
[21:52:57] LaTeX build process spawned. PID: 940.
[21:52:59] LaTeX log parsed with 0 messages.
[21:52:59] Recipe of length 1 finished. PID: 940.
[21:52:59] Successfully built d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex.
[21:52:59] No PDF viewer connected for d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.pdf

**Comment by joliang17** [comment]
This is the log for building tex file.

**Comment by joliang17** [comment]
![image](https://user-images.githubusercontent.com/47235014/66801385-c148f080-eecd-11e9-9f45-abe38cdec014.png)

Hold on. As the underline of the file path is wired in this picture, I removed the space in my file name, and build this tex file again. Then, .synctex file appears in my folder. 

So the space is not allowed in the file name?

**Comment by jlelong** [comment]
Spaces in filenames should not be a problem. The underlining of file paths is automatic and handled by vscode itself. It has nothing to do with how we treat paths in the extension.

Can you restore the spaces and show use the logs of the LaTeX compiler, not the ones of the extension?

**Comment by joliang17** [comment]
Sure. 
****************************************
The following log is for LaTeX workshop:

[15:18:02] Found root file from active editor: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[15:18:02] Root file remains unchanged from: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex.
[15:18:07] Parsing d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[15:18:07] File watcher: responding to change in d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[15:18:07] d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex changed. Auto build project.
[15:18:07] BUILD command invoked.
[15:18:07] Building root file: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[15:18:07] Build root file d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[15:18:07] Recipe step 1: pdflatex, --max-print-line=10000,-synctex=1,-interaction=nonstopmode,-file-line-error,d:/Google Drive/UCB/Fall 2019/242/hw3/Homework 3
[15:18:07] LaTeX build process spawned. PID: 18520.
[15:18:09] LaTeX log parsed with 0 messages.
[15:18:09] Recipe of length 1 finished. PID: 18520.
[15:18:09] Successfully built d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex.
[15:18:09] No PDF viewer connected for d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.pdf
[15:18:11] Found root file from active editor: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex
[15:18:11] Root file remains unchanged from: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex.
[15:19:12] Found root file from active editor: d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.tex

****************************************
The following log is for LaTeX compiler:

This is pdfTeX, Version 3.1415926-2.3-1.40.12 (MiKTeX 2.9)
entering extended mode
("d:/Google Drive/UCB/Fall 2019/242/hw3/Homework 3.tex"
LaTeX2e <2011/06/27>
Babel <v3.8m> and hyphenation patterns for english, afrikaans, ancientgreek, arabic, armenian, assamese, basque, bengali, bokmal, bulgarian, catalan, coptic, croatian, czech, danish, dutch, esperanto, estonian, farsi, finnish, french, galician, german, german-x-2009-06-19, greek, gujarati, hindi, hungarian, icelandic, indonesian, interlingua, irish, italian, kannada, kurmanji, lao, latin, latvian, lithuanian, malayalam, marathi, mongolian, mongolianlmc, monogreek, ngerman, ngerman-x-2009-06-19, nynorsk, oriya, panjabi, pinyin, polish, portuguese, romanian, russian, sanskrit, serbian, slovak, slovenian, spanish, swedish, swissgerman, tamil, telugu, turkish, turkmen, ukenglish, ukrainian, uppersorbian, usenglishmax, welsh, loaded.
(E:\Latex\miktex\tex/latex\base\article.cls
Document Class: article 2018/09/03 v1.4i Standard LaTeX document class
(E:\Latex\miktex\tex/latex\base\size11.clo)) (E:\Latex\CTEX\MiKTeX\tex\latex\booktabs\booktabs.sty) (E:\Latex\miktex\tex/latex\hyperref\hyperref.sty (E:\Latex\miktex\tex/generic\oberdiek\hobsub-hyperref.sty (E:\Latex\miktex\tex/generic\oberdiek\hobsub-generic.sty)) (E:\Latex\miktex\tex/latex\graphics\keyval.sty) (E:\Latex\miktex\tex/generic\ifxetex\ifxetex.sty) (E:\Latex\miktex\tex/latex\oberdiek\auxhook.sty) (E:\Latex\miktex\tex/latex\oberdiek\kvoptions.sty) (E:\Latex\miktex\tex/latex\hyperref\pd1enc.def) (E:\Latex\miktex\tex/latex\00miktex\hyperref.cfg) (E:\Latex\CTEX\MiKTeX\tex\latex\ltxmisc\url.sty)) (E:\Latex\miktex\tex/latex\hyperref\hpdftex.def (E:\Latex\miktex\tex/latex\oberdiek\rerunfilecheck.sty)) (E:\Latex\miktex\tex/latex\graphics\graphicx.sty (E:\Latex\miktex\tex/latex\graphics\graphics.sty (E:\Latex\miktex\tex/latex\graphics\trig.sty) (E:\Latex\miktex\tex/latex\graphics-cfg\graphics.cfg) (E:\Latex\miktex\tex/latex\graphics-def\pdftex.def))) (E:\Latex\miktex\tex/latex\amsmath\amsmath.sty
For additional information on amsmath, use the `?' option.
(E:\Latex\miktex\tex/latex\amsmath\amstext.sty (E:\Latex\miktex\tex/latex\amsmath\amsgen.sty)) (E:\Latex\miktex\tex/latex\amsmath\amsbsy.sty) (E:\Latex\miktex\tex/latex\amsmath\amsopn.sty)) (E:\Latex\miktex\tex/latex\amsfonts\amssymb.sty (E:\Latex\miktex\tex/latex\amsfonts\amsfonts.sty)) (E:\Latex\CTEX\MiKTeX\tex\latex\subfigure\subfigure.sty
****************************************
* Local config file subfigure.cfg used *
****************************************
(E:\Latex\CTEX\MiKTeX\tex\latex\subfigure\subfigure.cfg)) (E:\Latex\miktex\tex/latex\xcolor\xcolor.sty (E:\Latex\miktex\tex/latex\graphics-cfg\color.cfg)) (E:\Latex\miktex\tex/latex\commath\commath.sty (E:\Latex\miktex\tex/latex\base\ifthen.sty)) (E:\Latex\miktex\tex/latex\listings\listings.sty (E:\Latex\miktex\tex/latex\listings\lstmisc.sty) (E:\Latex\miktex\tex/latex\listings\listings.cfg)) (E:\Latex\miktex\tex/latex\listings\lstlang1.sty) (E:\Latex\miktex\tex/latex\listings\lstlang2.sty) (E:\Latex\miktex\tex/latex\listings\lstlang3.sty) ("d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.aux") (E:\Latex\miktex\tex/latex\hyperref\nameref.sty (E:\Latex\miktex\tex/generic\oberdiek\gettitlestring.sty)) ("d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.out") ("d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.out") (E:\Latex\CTEX\MiKTeX\tex\context\base\supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (E:\Latex\miktex\tex/latex\oberdiek\epstopdf-base.sty (E:\Latex\miktex\tex/latex\oberdiek\grfext.sty)) (E:\Latex\miktex\tex/latex\amsfonts\umsa.fd) (E:\Latex\miktex\tex/latex\amsfonts\umsb.fd)
SyncTeX warning: Synchronization was disabled from
the command line with -synctex=0
Changing the value of \synctex has no effect. [1{E:/Latex/CTEX/UserData/pdftex/config/pdftex.map{Unicode.sfd}{UGBK.sfd}}] [2] [3] ("d:\Google Drive\UCB\Fall 2019\242\hw3\Homework 3.aux") )
(see the transcript file for additional information)<E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmex10.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmmi10.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmmi6.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmmi8.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmr10.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmr12.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmr17.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmr6.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmr8.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmsy10.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmsy6.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/cm/cmsy8.pfb><E:/Latex/miktex/fonts/type1/public/amsfonts/symbols/msam10.pfb>
Output written on "Homework 3.pdf" (3 pages, 135987 bytes).
Transcript written on "Homework 3.log".

*************************************
- The json settings in vscode about LaTeX workshop:
```json
"latex-workshop.latex.recipes": [
        {
            "name": "pdflatex",
            "tools": [
            "pdflatex"
        ]
        },{
            "name": "xelatex",
            "tools": [
                "xelatex"
            ]
        }, {
            "name": "latexmk",
            "tools": [
                "latexmk"
            ]
        },
        {
            "name": "pdflatex -> bibtex -> pdflatex*2",
            "tools": [
                "pdflatex",
                "bibtex",
                "pdflatex",
                "pdflatex"
            ]
        }
    ],
    "latex-workshop.latex.tools": [{
            "name": "latexmk",
            "command": "latexmk",
            "args": [
                "-synctex=1",
                "-interaction=nonstopmode",
                "-file-line-error",
                "-pdf",
                "-outdir=%OUTDIR%",
                "%DOC%"
            ],
            "env": {}
        }, {
            "name": "xelatex",
            "command": "xelatex",
            "args": [
                "-synctex=1",
                "-interaction=nonstopmode",
                "-file-line-error",
                "%DOC%"
            ]
        },
        {
            "name": "pdflatex",
            "command": "pdflatex",
            "args": [
                "-synctex=1",
                "-interaction=nonstopmode",
                "-file-line-error",
                "%DOC%"
            ],
            "env": {}
        },
        {
            "name": "bibtex",
            "command": "bibtex",
            "args": [
                "%DOCFILE%"
            ],
            "env": {}
        }
    ],
```

**Comment by jlelong** [comment]
There is something strange in your logs because the extension logs says

```
Recipe step 1: pdflatex, --max-print-line=10000,-synctex=1,-interaction=nonstopmode,-file-line-error,d:/Google Drive/UCB/Fall 2019/242/hw3/Homework 3
```

and yet in the middle of the compiler logs you have

```
SyncTeX warning: Synchronization was disabled from the command line with -synctex=0
```

You may have something disabling synctex in your tex file. In any case, this is not related to the extension.

**Comment by joliang17** [comment]
Quite wired, as when I removed the space from the file name, everything goes well again. 

Thank you for following up this issues!

**Comment by trevorgunn** [comment]
I just tested this and I am seeing similar behaviour...on Windows. Linux seems to be fine.

Essentially, if the tex file has spaces in the name (spaces in the folder names are fine), then everything seems to be handled fine by the extension (the path is correct as I walk through the code), but somehow the `vscode.window.showTextDocument` function doesn't do its job.

**Comment by trevorgunn** [comment]
Update:

I tried duplicating the issue in a MWE extension and everything worked fine. Then, I went back to our code and started commenting out stuff and it seems that

https://github.com/James-Yu/LaTeX-Workshop/blob/4e381f056af278fd7ec158386ea36d6855b8cd8b/src/components/locator.ts#L328

is the offending line.

I am not sure why this line causes issues if the file has spaces in it. I am also not sure what the line is supposed to accomplish. Does someone else know more?
